Specification.md

```diff
@@ -92,12 +92,31 @@ they also MUST NOT only consist of “blank characters”.
 #### Tag
 The purpose of *tags* is to help categorise *records* and *entries*.
 
-> Examples: `#gym`, `#24hours`, `#home-office`, `#読む`, `#ticket_127`
+> Examples: `#gym`, `#24hours`, `#home-office`, `#読む`, `#ticket=891`, `#song="Yellow Submarine"`
 
 Any amount of *tags* MAY appear anywhere within *summaries*.
 
-A *tag* MUST only contain “letters”, “digits”, or the characters `_` or `-`.
-It MUST be preceded by a single `#` character.
+A *tag* MUST consist of a *tag name*,
+which MUST be preceded by a single `#` character.
+The *tag name* MAY be followed by a `=` character
+and a *tag value*.
+
+The *tag name* MUST only contain
+“letters”, “digits”, or the characters `_` or `-`.
+It MUST be interpreted as if it was all lower-case.
+
+The *tag value* MAY be surrounded by a pair of matching quotes,
+which MUST either be `"` (RECOMMENDED) or `'`.
+- If the *tag value* is quoted, it MAY contain any character
```

Comment:
Could this please be clarified?

Should this be `any "letter"`, or does it literally mean *any* character?

Reply:
Side note: are emojis allowed?

Reply:
The idea was that quoted values can contain *any* character, e.g.

```
#project="56/12.3"
#ticket="#169"
#foo="!&^@#&*♥½⅓ä€¥®!"
#movie="Matrix 4"
#mood="🥴"
```

In the above cases, the quoting mechanism would be necessary, otherwise `#project=56/12.3` would only be partially recognised as `#project=56`.

My reasoning behind this was that the tag *value* likely references something external, such as a ticket or project id, which the user should be able to “cite” literally. That’s in contrast to the tag *name*, which is more supposed to be a logical keyword, rather than a strict, literal token.

The default way should be for tag values to be unquoted, just because it’s simpler. The quoting is only so that users don’t have to convert special characters, which might be annoying – think, their canonical project id is usually `56/12.3`, then they’d have to convert that to `#project=56-12-3`, but with quoting, they can do `#project="56/12.3"`.

An example for a use case is https://github.com/jotaen/klog/discussions/55. The [wildcard filtering feature](https://github.com/jotaen/klog/pull/75) of the CLI was introduced as workaround for that, but I think the proper solution is tag values.

Does that make sense?

Reply:
That does make sense! Thank you for clarifying :)

Reply:
Thanks for checking this out!

Reply:
Btw., the main branch already contains [the implementation](https://github.com/jotaen/klog/pull/180) for this, just in case you want to play around with it.

Reply:
Will use that when implementing syntax highlighting!